src/constants/signUp.validation.ts

```diff
@@ -21,5 +29,11 @@ export const signUpSchema = yup.object().shape({
     .matches(uppercaseRegex, validationErrors.PASSWORD_INVALID_VALUE(`${passwordUppercaseLettersAmount} uppercase letters`))
     .matches(numbersRegex, validationErrors.PASSWORD_INVALID_VALUE(`${passwordNumbersAmount} numbers`))
     .matches(symbolsRegex, validationErrors.PASSWORD_INVALID_VALUE(`${passwordSymbolsAmount} symbols`))
+    .required(validationErrors.FIELD_REQUIRED),
+  confirmPassword: yup
+    .string()
+    .oneOf([yup.ref('password'), undefined], validationErrors.PASSWORDS_MUST_MATCH)
     .required(validationErrors.FIELD_REQUIRED)
 });
+
+export interface ISignUp extends yup.InferType<typeof signUpSchema> {}
```

Comment:
Please investigate `yup.InferType<typeof signUpSchema>`.Why do you need `SignUpSchemaInterface`? And this extending is unneeded  at least for now. Just the schema type.

Reply:
Changed on:
```
export type SignUpSchemaType = yup.InferType<typeof signUpSchema>;
```